src/http/client/response.cr

```diff
@@ -3,20 +3,33 @@ require "../common"
 
 class HTTP::Client::Response
   getter version : String
-  getter status_code : Int32
-  getter status_message : String
+  getter status : HTTP::Status
+  getter status_message : String?
   getter headers : Headers
   getter! body_io : IO
   @cookies : Cookies?
 
-  def initialize(@status_code, @body : String? = nil, @headers : Headers = Headers.new, status_message = nil, @version = "HTTP/1.1", @body_io = nil)
-    @status_message = status_message || HTTP.default_status_message_for(@status_code)
+  def initialize(@status : HTTP::Status, @body : String? = nil, @headers : Headers = Headers.new, status_message = nil, @version = "HTTP/1.1", @body_io = nil)
+    @status_message = status_message || @status.description
 
-    if Response.mandatory_body?(@status_code)
+    if Response.mandatory_body?(@status)
       @body = "" unless @body || @body_io
     else
       if (@body || @body_io) && (headers["Content-Length"]? != "0")
-        raise ArgumentError.new("Status #{status_code} should not have a body")
+        raise ArgumentError.new("Status #{status.code} should not have a body")
+      end
+    end
+  end
+
+  def initialize(status_code : Int32, @body : String? = nil, @headers : Headers = Headers.new, status_message = nil, @version = "HTTP/1.1", @body_io = nil)
```

Comment:
This should just be a `.new` override accepting Int32 and delegating to the automatically defined `.new` method of the previously declared constructor:
```cr
def self.new(status_code : Int32 body : String? = nil, headers : Headers = Headers.new, status_message = nil, version = "HTTP/1.1", body_io = nil)
  new(HTTP::Status.new(status_code), body, headers, status_message, version, body_io)
end
```
There is no need to duplicate the entire logic.